language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java

```diff
@@ -276,6 +277,20 @@ public static String genSortTextByAssignability(BallerinaCompletionContext conte
      * @return True if assignable
      */
     public static boolean isCompletionItemAssignable(LSCompletionItem completionItem, TypeSymbol typeSymbol) {
+        if (typeSymbol.typeKind() == TypeDescKind.TYPEDESC && completionItem.getType() == SYMBOL) {
+            Optional<Symbol> optionalSymbol = ((SymbolCompletionItem) completionItem).getSymbol();
+            if (optionalSymbol.isPresent()) {
+                if (optionalSymbol.get().kind() == SymbolKind.TYPE_DEFINITION
+                        || optionalSymbol.get().kind() == SymbolKind.TYPE) {
+                    TypeDescTypeSymbol typeDescTypeSymbol = (TypeDescTypeSymbol) typeSymbol;
+                    if (typeDescTypeSymbol.typeParameter().isPresent()) {
+                        Optional<TypeSymbol> optionalTypeSymbol = getSymbolFromCompletionItem(completionItem);
+                        return optionalTypeSymbol.isPresent()
+                                && optionalTypeSymbol.get().subtypeOf(typeDescTypeSymbol.typeParameter().get());
+                    }
+                }
+            }
+        }
         if (completionItem.getCompletionItem().getKind() == CompletionItemKind.TypeParameter) {
```

Comment:
Let's remove this condition and try to capture TypeParameters in line 284 as well

Reply:
@malinthar  If this goes to line 284, completionItem should be a `Symbol`. There are cases where this cannot be a `Symbol` thats why kept it here.

Reply:
> @malinthar If this goes to line 284, completionItem should be a `Symbol`. There are cases where this cannot be a `Symbol` thats why kept it here.

Will you elaborate with an example please

Reply:
As you have mentioned we have only considered Symbol type completion items at the moment. However there are some completions items that are not symbol typed. Their type must be Snippet. However, their kind is TypeParamater. One such completion item is the completion item with label Thread. The idea is to capture the TypeParamater(kind) completion items as assignable when the expected type is TypeDesc. But according to the existing logic we always return not assignable for such completion items. Shall we try to come up with a logic to capture that as well.

Reply:
@malinthar I mean there can be `CompletionItem` with its kind being a `TypeParameter` and its type not being a `Symbol` where it can be `Snippet`. I didn't find any example. Is it possible?

Reply:
@malinthar I'll do.